src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java

```diff
@@ -449,6 +466,125 @@ private boolean checkDependencies() throws MojoExecutionException {
         return warning;
     }
 
+    // todo: enhance analyzer (dependency) to do it since it already visits classes
+    //       will save some time
+    private Set<String> scanForSpiUsage(
+            final Set<Artifact> usedDeclared, final Map<Artifact, Set<String>> usedUndeclaredWithClasses) {
+        return Stream.concat(
+                        usedDeclared.stream().flatMap(this::findUsedSpi),
+                        usedUndeclaredWithClasses.keySet().stream().flatMap(this::findUsedSpi))
+                .collect(toSet());
+    }
+
+    private Stream<String> findUsedSpi(final Artifact artifact) {
```

Comment:
Can we extends a [maven-dependency-analyzer](https://github.com/apache/maven-dependency-analyzer) with such feature?